Improve directory listings documentation (#5259)
mintlify/docs: organize/navigation.mdx

### Directory listings

> Use the `directory` property to automatically render a directory of child pages on group root pages. When you set `directory` on a navigation node, any group with a `root` page beneath that node displays a listing of its children below the page content.

Reviewer: ```suggestion
Use the `directory` property to automatically render a directory of child pages on group root pages. When you set `directory` on a navigation element, any group with a `root` page beneath that element displays a listing of its pages below the page content.
```

> Use the `directory` property to automatically render a directory of child pages on group root pages. When you set `directory` on a navigation element, any group with a `root` page beneath that element displays a listing of its pages below the page content.

Reviewer 2: ```suggestion
Use the `directory` property to automatically render a directory of child pages and groups on group root pages. When you set `directory` on any object within the `navigation` tree in `docs.json`, groups with a `root` page matching or beneath that object display a listing of their pages and groups below their page contents.
```

The `directory` property accepts three values:

| Value | Behavior |
| :---------- | :------- |

> | `"none"` | No directory listing. This is the default. |

Reviewer: ```suggestion
| `"none"` | No directory listing. Default value. |
```

> | `"accordion"` | Displays children in a collapsible list grouped by section. |

Reviewer: ```suggestion
| `"accordion"` | Displays child pages in a collapsible list grouped by section. |
```

> | `"card"` | Displays children in a horizontal card layout. |

Reviewer: ```suggestion
| `"card"` | Displays child pages in a horizontal card layout. |
```

> The `directory` value inherits recursively through the navigation tree. Set it on a parent node and all descendant groups inherit the same value. Any descendant can override the inherited value, including setting `"none"` to opt out.

Reviewer: ```suggestion
The `directory` value inherits recursively through the navigation tree. Set it on a parent navigation element and all descendant groups inherit the same value. Any descendant can override the inherited value by setting `"directory": "none"`.
```

> The `directory` value inherits recursively through the navigation tree. Set it on a parent navigation element and all descendant groups inherit the same value. Any descendant can override the inherited value by setting `"directory": "none"`.

Reviewer 2: ```suggestion
The `directory` value inherits recursively through the navigation tree. Set it on anywhere within the navigation object and all descendant groups inherit the same value. Any descendant can override the inherited value by setting `directory` to a different value.
```
